app/src/main/java/tech/relaycorp/gateway/domain/publicsync/DeliverParcelsToGateway.kt

```diff
@@ -54,7 +54,9 @@ class DeliverParcelsToGateway
                 try {
                     deliverParcel(poWebClient, parcel)
                 } catch (e: ServerException) {
-                    logger.log(Level.INFO, "Could not deliver parcels due to server error", e)
+                    logger.log(Level.INFO, "Could not deliver parcel due to server error", e)
+                } catch (e: Exception) {
```

Comment:
Same comments as above, re: parcel collections (and apologies that I missed this catch for `ServerException` before!).

Now that I'm looking at this, when is the parcel deleted after delivery? In addition to deleting the parcel upon successful delivery, I think we should also delete when we get a `ForbiddenException`: That means that the Android Gateway isn't authorised to deliver parcels to that public gateway. And even if the Android GW re-registered, the parcel would still be using the old certificate chain, so the parcel could never be delivered.

Finally, can you please add tests for these changes?

Reply:
Forbidden exception is handled by the library and thrown as a RejectedParcelException. And we're already handling that, deleting the parcel and testing it.

Reply:
Good point re: deleting the parcel! In that case, my only suggestion here is to use `PDCException` instead of `Exception`